### ISSUE TITLE
Documentation enhancement: Proxmox Builder

### ISSUE THREAD

**Issue by udf2457** [hashicorp/packer-plugin-proxmox#184]
Dear Packer Team,

Re: Your Proxmox Builder Docs (https://www.packer.io/docs/builders/proxmox.html)

It would be nice if you actually documented precisely what permissions Packer's Proxomox builder is expecting.

As you know Proxomox allows granular definition of user privileges (https://pve.proxmox.com/wiki/User_Management)

As you also know, its 2019 ... and so "principle of least privilege" is king, not "meh, just give it god rights". ;-)

TL;DR ...  Does the Proxmox builder really need `Sys.PowerMgmt` , `Sys.Console`, `User.Modify` etc. etc. etc.

**Comment by hamannju** [comment]
Hello, so I just did this configuration on my cluster and settled on the following config: 

- Create a new resource pool "packer"
- Create a new group "packer_group"
- Create a new user "packer" and add it to "packer_group"
- Add the desired datastore to "packer" resource pool
- Assign the following privileges to packer_group:
  - PVEPoolUser on /pools/packer
  - PVEDatastoreUser on /pools/packer
  - PVEVMAdmin on /pools/packer

If a separate datastore for images is used also the following is needed:
- PVEDatastoreAdmin on /storage/iso-storage

This is because packer deletes the ephemeral images on this datastore at the end of its run which requires the Datastore.Allocate permission which the normal DatastoreUser does not have.

If you follow this setup you will have VM Pool where the packer user can create and delete only its own VMs which are isolated from the rest of the cluster. I think this should be the least amount of privileges to be able to execute all necessary packer tasks.

I followed this discussion on the Proxmox forum for this setup: https://forum.proxmox.com/threads/allow-user-to-create-vms-but-only-see-and-manage-those-created-by-itself.121222/
